modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -242,6 +242,9 @@ def split_partitions_using_pivots_for_sort(
     tuple[pandas.DataFrame]
         A tuple of the splits from this partition.
     """
+    if len(pivots) == 0:
+        # We can return the dataframe with zero changes if there were no pivots passed
+        return (df,)
```

Comment:
2nd safeguard:
If we somehow proceeded to the splitting logic having no pivot values just return the exact same frame we received as an input